python/arcticdb/adapters/mongo_library_adapter.py

```diff
@@ -0,0 +1,102 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+from pickletools import int4
+import re
+import time
+from typing import Optional
+
+from arcticdb.options import LibraryOptions
+from arcticc.pb2.storage_pb2 import EnvironmentConfigsMap, LibraryConfig
+from arcticdb.version_store.helper import add_mongo_library_to_env
+from arcticdb.config import _DEFAULT_ENV
+from arcticdb.version_store._store import NativeVersionStore
+from arcticdb.adapters.arctic_library_adapter import ArcticLibraryAdapter, set_library_options
+from arcticdb_ext.storage import Library
+from arcticdb.encoding_version import EncodingVersion
+from collections import namedtuple
+from dataclasses import dataclass, fields
+
+PARSED_QUERY = namedtuple("PARSED_QUERY", ["region"])
+
+
+@dataclass
+class ParsedQuery:
+    min_pool_size: Optional[int] = 0
+    max_pool_size: Optional[int] = 0
+    selection_timeout_ms: Optional[int] = 0
+
+
+class MongoLibraryAdapter(ArcticLibraryAdapter):
+    REGEX = r"(?P<endpoint>mongodb://[^?]+)(?P<options>\?.*)?"
+
+    @staticmethod
+    def supports_uri(uri: str) -> bool:
+        return uri.startswith("mongodb://")
+
+    def __init__(self, uri: str, *args, **kwargs):
+        self._uri = uri
+        match = re.match(self.REGEX, uri)
+
+        option_regex = r"(?P<option>[^=]+)=(?P<value>.*)"
+        additional_options = dict()
+        self._endpoint = match["endpoint"]
+        if match["options"] != None:
+            additional_options = self._parse_query(match["options"])
+            self._endpoint += "&".join(
+                [
+                    f"{match[0]}={match[1]}"
+                    for option_str in match["options"].lstrip("?").split("&")
+                    for match in re.findall(option_regex, option_str)
+                    if match[0] not in additional_options
+                ]
+            )
+        self._query_params: ParsedQuery = ParsedQuery(**additional_options)
+        self._min_pool_size = self._query_params.min_pool_size
+        self._max_pool_size = self._query_params.max_pool_size
+        self._selection_timeout_ms = self._query_params.selection_timeout_ms
+
+        super().__init__(uri, EncodingVersion.V1)
+
+    def __repr__(self):
+        return (
+            f"mongodb(endpoint={self._endpoint}, min_pool_size={self._min_pool_size},"
```

Comment:
`self._endpoint` may contain authentication info, which should be obfuscated in the repo. Again hopefully `pymongo` can help with this, we shouldn't have to write it ourselves.

Reply:
I just strip the credential in the function `__repr__`